nexus/src/app/instance.rs

```diff
@@ -74,10 +78,10 @@ impl super::Nexus {
                         .await?;
                 Ok(authz_instance.id())
             }
-            params::InstanceSelector::InstanceProjectAndOrgId {
-                instance_name,
-                project_name,
-                organization_id,
+            params::InstanceSelector {
+                instance: NameOrId::Name(instance_name),
+                project: Some(NameOrId::Name(project_name)),
+                organization: Some(NameOrId::Id(organization_id)),
```

Comment:
I see, tbe optional parent identifiers aren’t a problem because you were already doing this whole pattern match on the valid combinations anyway. So encoding them into the type was redundant and way harder for spec generation.